identity-core/src/crypto/merkle_key/verifier.rs

```diff
@@ -183,35 +177,28 @@ fn expand_signature_value(
     .map(Into::into)?;
 
   // Extract bytes of the base58-encoded proof
-  let proof: Vec<u8> = utils::decode_b58(proof).map_err(|_| errors::InvalidProofFormat)?;
+  let proof: Vec<u8> = utils::decode_b58(proof).map_err(|_| InvalidProofFormat)?;
 
   // Decode the signature value for the underlying signature implementation
-  let signature: Vec<u8> =
-    utils::decode_b58(signature).map_err(|_| VerificationProcessingError::from("failed to parse signature"))?;
+  let signature: Vec<u8> = utils::decode_b58(signature)
+    .map_err(|_| VerificationError::ProcessingFailed("failed to parse signature".into()))?;
 
   Ok((public, proof, signature))
 }
 
 mod errors {
   use crate::crypto::merkle_key::MerkleKeyTagExtractionError;
   use crate::crypto::signature::errors::VerificationError;
-  use crate::crypto::signature::errors::VerificationProcessingError;
 
   pub(super) struct InvalidProofFormat;
 
-  impl From<InvalidProofFormat> for VerificationProcessingError {
-    fn from(_: InvalidProofFormat) -> Self {
-      Self::from("invalid proof format")
-    }
-  }
-
   impl From<InvalidProofFormat> for VerificationError {
-    fn from(err: InvalidProofFormat) -> Self {
-      Self::ProcessingFailed(err.into())
+    fn from(_: InvalidProofFormat) -> Self {
+      Self::ProcessingFailed("invalid proof format".into())
```

Comment:
Mentioning this because I know you like optimizations. Using `into` means it's turned into a `Cow` at runtime, while Rust would probably be able to evaluate this at compile-time, if you use the slightly less ergonomic `Cow::Borrowed("invalid proof format")`.